rotkehlchen/chain/evm/decoding/velodrome/velodrome_cache.py

```diff
@@ -78,41 +83,81 @@ def save_velodrome_data_to_cache(
+            gauge_key = (CacheType.VELODROME_GAUGE_ADDRESS,)
+        else:
+            pool_key = (CacheType.AERODROME_POOL_ADDRESS,)
+            gauge_key = (CacheType.AERODROME_GAUGE_ADDRESS,)
+
         globaldb_set_general_cache_values(
             write_cursor=write_cursor,
-            key_parts=(CacheType.VELODROME_POOL_ADDRESS,),
+            key_parts=pool_key,  # type: ignore
             values=[pool.pool_address],
         )
         if pool.gauge_address is not None:
             globaldb_set_general_cache_values(
                 write_cursor=write_cursor,
-                key_parts=(CacheType.VELODROME_GAUGE_ADDRESS,),
+                key_parts=gauge_key,  # type: ignore
                 values=[pool.gauge_address],
             )
 
 
-def read_velodrome_pools_and_gauges_from_cache() -> tuple[set[ChecksumEvmAddress], set[ChecksumEvmAddress]]:  # noqa: E501
+def read_velodrome_like_pools_and_gauges_from_cache(
+        counterparty: Literal['velodrome', 'aerodrome'],
+) -> tuple[set[ChecksumEvmAddress], set[ChecksumEvmAddress]]:
     """
     Reads globaldb cache and returns:
-    - A set of all known addresses of velodrome pools.
-    - A set of all known addresses of velodrome gauges.
+    - A set of all known addresses of (velo/aero)drome pools.
+    - A set of all known addresses of (velo/aero)drome gauges.
     """
     gauge_addresses = set()
     pool_addresses = set()
+    if counterparty == CPT_VELODROME:
+        cache_type_pool: Literal[
+            CacheType.VELODROME_POOL_ADDRESS,
+            CacheType.AERODROME_POOL_ADDRESS,
+        ] = CacheType.VELODROME_POOL_ADDRESS
+        cache_type_gauge: Literal[
+            CacheType.VELODROME_GAUGE_ADDRESS,
+            CacheType.AERODROME_GAUGE_ADDRESS,
+        ] = CacheType.VELODROME_GAUGE_ADDRESS
+    else:
+        cache_type_pool = CacheType.AERODROME_POOL_ADDRESS
+        cache_type_gauge = CacheType.AERODROME_GAUGE_ADDRESS
+
     with GlobalDBHandler().conn.read_ctx() as cursor:
-        str_pool_addresses = globaldb_get_general_cache_values(cursor=cursor, key_parts=(CacheType.VELODROME_POOL_ADDRESS,))  # noqa: E501
+        str_pool_addresses = globaldb_get_general_cache_values(cursor=cursor, key_parts=(cache_type_pool,))  # noqa: E501
         for pool_address in str_pool_addresses:
             pool_addresses.add(string_to_evm_address(pool_address))
 
-        str_gauge_addresses = globaldb_get_general_cache_values(cursor=cursor, key_parts=(CacheType.VELODROME_GAUGE_ADDRESS,))  # noqa: E501
+        str_gauge_addresses = globaldb_get_general_cache_values(cursor=cursor, key_parts=(cache_type_gauge,))  # noqa: E501
         for gauge_address in str_gauge_addresses:
             gauge_addresses.add(string_to_evm_address(gauge_address))
 
     return pool_addresses, gauge_addresses
 
 
+def read_velodrome_pools_and_gauges_from_cache() -> tuple[set[ChecksumEvmAddress], set[ChecksumEvmAddress]]:  # noqa: E501
```

Comment:
Both these functions are not really needed.

Can't you just call the one you need with the appropriate argument?

Reply:
Can you write a comment in the docstring why you need such a simple function. And this is a quite bad design ...